### PR TITLE
fix: request permissions before using camera/media library in image picker

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
What it says on the tin. In draft til I can validate on iOS. Right now when you fix the permissions in settings from the slope screen and then return to the app, you need to leave/re-enter the slope screen to get the app to pick up on the change, the `usePermissions` hook doesn't seem to re-render the screen. not sure if there's a better workaround.

### Checklist
- [x] Corresponding issue has been opened

### Related Issues
Fixes https://github.com/techmatters/terraso-mobile-client/issues/1328

### CODE REVIEW

File: dev-client/src/components/inputs/image/ImagePicker.tsx
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/ImagePicker.tsx:66]:
```suggestion
  const [cameraPermission, requestCameraPermission] = useCameraPermissions();
  const [mediaPermission, requestMediaPermission] =
```

File: dev-client/src/components/inputs/image/ImagePicker.tsx
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/ImagePicker.tsx:71]:
```suggestion
      cameraPermission,
```

File: dev-client/src/components/inputs/image/ImagePicker.tsx
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/ImagePicker.tsx:86]:
```suggestion
  }, [cameraPermission, requestCameraPermission, onPick]);
```

File: dev-client/src/components/inputs/image/ImagePicker.tsx
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/ImagePicker.tsx:98]:
```suggestion
  }, [mediaPermission, requestMediaPermission, onPick]);
```

File: dev-client/src/components/inputs/image/callWithPermissions.ts
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/callWithPermissions.ts:4]:
```suggestion
export const callWithPermission = async (
  permission: PermissionResponse | null,
```

File: dev-client/src/components/inputs/image/callWithPermissions.ts
Comment (paulschreiber) [inline dev-client/src/components/inputs/image/callWithPermissions.ts:8]:
```suggestion
  if (permission?.granted) {
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: dev-client/src/App.tsx
Comment (shrouxm) [inline dev-client/src/App.tsx:80]: comment: @tm-ruxandra i got an app crash due to `BottomSheetModalProvider` being in the wrong place, this change is just a quick fix to be able to validate this fix but i wanted to check if you ever did any more work on this or if i'm good to do the fix we talked about (moving it into screen scaffold, below the header height provider) without stepping on your toes

Comment (tm-ruxandra) [reply dev-client/src/App.tsx:80]: @shrouxm Totally fine if you need to make this change here, it won't conflict with any of my ongoing work.

**Review by tm-ruxandra — APPROVED** [review]

**Comment by paulschreiber** [comment]
@shrouxm does this need any more work before it can be merged?

File: dev-client/src/translations/en.json
Comment (paulschreiber) [inline dev-client/src/translations/en.json:448]: s/LandPKS/LandPKS Soil ID/

File: dev-client/src/translations/en.json
Comment (paulschreiber) [inline dev-client/src/translations/en.json:446]: s/LandPKS/LandPKS Soil ID/

**Comment by shrouxm** [comment]
@paulschreiber it looks like the issues that were blocking merging this got fixed at some point so should be good to go once you approve!

File: dev-client/src/components/modals/PermissionsRequestModal.tsx
Comment (paulschreiber) [inline dev-client/src/components/modals/PermissionsRequestModal.tsx:51]: question: can we check for any falsy value here, i.e. `!permissions`?

**Review by paulschreiber — APPROVED** [review]

File: dev-client/src/components/modals/PermissionsRequestModal.tsx
Comment (shrouxm) [reply dev-client/src/components/modals/PermissionsRequestModal.tsx:51]: the type of `permissions` is `PermissionsResponse | null` so it can't hold any other falsy value. if in the future it can hold a falsy value, we might want to handle it differently so i think it's good the type system will notify us that that's the case

**Pull request merged**
